### PR TITLE
fix(transmission): Move logs out of syslog

### PR DESCRIPTION
**Pull request by flying-sausages**
## Description

systems with transmission can have an unresponsive `journalctl` due to the quantity of logs from torrents with errors in a session being spammed into syslog

## Fixes issues: 
- `journalctl`/`systemctl` being slow on systems with transmission

## Proposed Changes:
- Move logs to a dedicated file

## Change Categories

- Bug fix 

## Checklist

- [x] Docs have been made OR are not necessary
    - PR link: 
- [x] Changes to panel have been made OR are not necessary
    - PR link: 
- [x] Code is formatted [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Shellcheck isn't screaming [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#editor-plugins-and-tooling)
- [x] Prints to terminal are handled [(See more)](https://github.com/swizzin/swizzin/blob/master/CONTRIBUTING.md#printing-into-the-terminal)
- [x] I have commented my code, particularly in hard-to-understand areas

### Architectures

|   			| `amd64` 	| `armhf` 	| `arm64` 	| Unspecified 	|
|--------		|-------- 	|-------- 	|-------- 	|----------		|
| Focal 		|			|			|			|	🛠			|
| Bionic		|			|			|			|				|
| Buster		|			|			|			|				|
| Stretch		|			|			|			|				|
| Raspbian  	|	⚫️		|			|	⚫️		|	⚫️			|

### 🛠🛠 TODO
- Test install from scratch (MU/single does not matter)
- Test update on existing install (MU/Single does not matter)

### CODE REVIEW

File: scripts/update/transmission.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+if [ -f /install/.transmission.lock ]; then
+
+    if ! grep -q /home/systemd/system/transmission@.service "--logfile"; then
```
Comment (liaralabs) [inline scripts/update/transmission.sh:5]:
```suggestion
    if ! grep -q "\-\-logfile" /etc/systemd/system/transmission@.service ; then
```

File: scripts/update/transmission.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        echo_progress_start "Moving transmission's logs to a file instead of syslog"
+        : #switch service to logfile
```
Comment (liaralabs) [inline scripts/update/transmission.sh:7]: Do we need the colon?

File: scripts/update/transmission.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        sed "s/ExecStart=.*/ExecStart=/usr/bin/transmission-daemon -f --log-error --logfile /home/%i/.config/transmission-daemon/transmission.log/g" -i
```
Comment (liaralabs) [inline scripts/update/transmission.sh:8]: This is not the same command as the install script and also the forward slashes are going to cause this command to fail.

File: scripts/update/transmission.sh (hunk 2 above)
Comment (flying-sausages) [reply scripts/update/transmission.sh:7]: nope lol
```suggestion
```

File: scripts/update/transmission.sh (hunk 3 above)
Comment (flying-sausages) [reply scripts/update/transmission.sh]: I don't see the difference, can you point it out please?

Comment (liaralabs) [reply scripts/update/transmission.sh]: Oh I may have conflated the `-i` on the end for sed as part of the `ExecStart` command

File: scripts/update/transmission.sh [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+if [ -f /install/.transmission.lock ]; then
+
+    if ! grep -q "\-\-logfile" /etc/systemd/system/transmission@.service; then
+        echo_progress_start "Moving transmission's logs to a file instead of syslog"
+        sed "s|ExecStart=.*|ExecStart=/usr/bin/transmission-daemon -f --log-error --logfile /home/%i/.config/transmission-daemon/transmission.log|g" -i
```
Comment (liaralabs) [inline scripts/update/transmission.sh:7]:
```
sed: no input files
```

File: scripts/update/transmission.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+        systemctl daemon-reload
+        readarray -t users < <(_get_user_list)
```
Comment (liaralabs) [inline scripts/update/transmission.sh:9]:
```
/etc/swizzin/scripts/update/transmission.sh: line 9: _get_user_list: command not found
```

**Review by liaralabs — CHANGES_REQUESTED** [review]

**Comment by flying-sausages** [comment]
image

**Review by liaralabs — APPROVED** [review]

**Pull request merged**